pkg/components/metallb/component_test.go

```diff
@@ -97,3 +107,197 @@ component "metallb" {
 `
 	testRenderManifest(t, configHCL)
 }
+
+func getSpeakerDaemonset(t *testing.T, m map[string]string) *appsv1.DaemonSet {
+	dsStr, ok := m["daemonset-speaker.yaml"]
+	if !ok {
+		t.Fatalf("speaker daemonset config not found")
+	}
+
+	ds := &appsv1.DaemonSet{}
+	if err := yaml.Unmarshal([]byte(dsStr), ds); err != nil {
+		t.Fatalf("failed unmarshaling manifest: %v", err)
+	}
+
+	return ds
+}
+
+func getDeployController(t *testing.T, m map[string]string) *appsv1.Deployment {
+	deployStr, ok := m["deployment-controller.yaml"]
+	if !ok {
+		t.Fatalf("controller deployment config not found")
+	}
+
+	deploy := &appsv1.Deployment{}
+	if err := yaml.Unmarshal([]byte(deployStr), deploy); err != nil {
+		t.Fatalf("failed unmarshaling manifest: %v", err)
+	}
+
+	return deploy
+}
+
+// nolint:funlen
+func TestConversion(t *testing.T) {
+	configHCL := `
+component "metallb" {
+  address_pools = {
+	default = ["1.1.1.1/32", "2.2.2.2/32"]
+  }
+
+  speaker_toleration {
+	key      = "speaker_key1"
+	operator = "Equal"
+	value    = "value1"
+  }
+
+  speaker_toleration {
+	key      = "speaker_key2"
+    operator = "Equal"
+	value    = "value2"
+  }
+
+  speaker_node_selectors = {
+    "speaker_node_key1" = "speaker_node_value1"
+    "speaker_node_key2" = "speaker_node_value2"
+  }
+
+  controller_toleration {
+	key 	 = "controller_key1"
+	operator = "Equal"
+	value 	 = "value1"
+  }
+
+  controller_toleration {
+	key 	 = "controller_key2"
+	operator = "Equal"
+	value 	 = "value2"
+  }
+
+  controller_node_selectors = {
+    "controller_node_key1" = "controller_node_value1"
+    "controller_node_key2" = "controller_node_value2"
+  }
+
+  service_monitor = true
+}`
+
+	testRenderManifest(t, configHCL)
+	m := renderManifest(t, configHCL)
```

Comment:
Why can't we make `renderManifest` to check for empty manifests? `testRenderManifest` name sounds odd to me.

Reply:
`testRenderManifest` was the original function called.

Reply:
In this case we call `renderManifest` twice. That doesn't seem to make sense to me.

Reply:
ok then I will remove `testRenderManifest` from here.